src/EFCore.SqlServer/Query/Internal/Translators/SqlServerTimeOnlyMethodTranslator.cs

```diff
@@ -62,10 +62,11 @@ public SqlServerTimeOnlyMethodTranslator(ISqlExpressionFactory sqlExpressionFact
 
         if ((method == FromDateTime || method == FromTimeSpan)
             && instance is null
-            && arguments.Count == 1
-            && arguments[0].Type != typeof(TimeSpan))
+            && arguments.Count == 1)
         {
-            return _sqlExpressionFactory.Convert(arguments[0], typeof(TimeOnly));
+            return arguments[0].Type is typeof(TimeSpan)
```

Comment:
my bad, my suggestion does not even compile 😅

Reply:
I believe that 
```suggestion
            return arguments[0].Type == typeof(TimeSpan)
```
might build... I will try it out tomorrow, so don't trust me on this (well, not yet 😇 )

Reply:
It builds & passes the tests 🚀 (at least locally 😅 )

Reply:
but those 3 tests require updates to the baselines in several files:
 - test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerTest.cs
 - test/EFCore.SqlServer.FunctionalTests/Query/TPCGearsOfWarQuerySqlServerTest.cs
 - test/EFCore.SqlServer.FunctionalTests/Query/TPTGearsOfWarQuerySqlServerTest.cs
 - test/EFCore.SqlServer.FunctionalTests/Query/TemporalGearsOfWarQuerySqlServerTest.cs